benchmark/scripts/find_heap_bound.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+# See LICENSE for license details.
+
+import re
+import argparse
+from typing import NamedTuple
+import logging
+
+from monitor_job import monitor_job, JobFailedError
+
+BaseHeapSize = NamedTuple('JavaHeapSize', [('value', int), ('suffix', str)])
+class HeapSize(BaseHeapSize):
+    K_FACTOR = 1024
+    M_FACTOR = 1024*1024
+    G_FACTOR = 1024*1024*1024
+
+    def toBytes(self) -> int:
+        if self.suffix == "":
+            factor = 1
+        elif self.suffix == "K":
+            factor = self.K_FACTOR
+        elif self.suffix == "M":
+            factor = self.M_FACTOR
+        elif self.suffix == "G":
+            factor = self.G_FACTOR
+        return self.value * factor
+
+    def round_to(self, target):
+        """Round to positive multiple of target, only if we have the same suffix"""
+        if self.suffix == target.suffix:
+            me = self.toBytes()
+            tar = target.toBytes()
+            res = tar * round(me / tar)
+            if res == 0:
+                res = tar
+            return HeapSize.from_bytes(res)
+        else:
+            return self
+
+    def __truediv__(self, div):
+        b = self.toBytes()
+        res = int(b / div)
+        return HeapSize.from_bytes(res)
+
+    def __mul__(self, m):
+        b = self.toBytes()
+        res = int(b * m)
+        return HeapSize.from_bytes(res)
+
+    def __add__(self, rhs):
+        return HeapSize.from_bytes(self.toBytes() + rhs.toBytes())
+
+    def __sub__(self, rhs):
+        return HeapSize.from_bytes(self.toBytes() - rhs.toBytes())
+
+    # Is this useful?
+    def approach(self, target, min_step):
+        """Binary search approach target with minimum step"""
+        me = self.toBytes()
+        t = target.toBytes()
+        s = min_step.toBytes()
+
+        diff = abs(me - t)
+        step = int(diff / 2)
+        if me > t:
+            res = me - step
+        else:
+            res = me + step
+        # Round to nearest multiple of s for prettiness
+        res = s * round(res / s)
+        # And saturate at s
+        res = s if res == 0 else res
+        return HeapSize.from_bytes(res)
+
+
+    @classmethod
+    def from_str(cls, s: str):
+        regex = '(\d+)([kKmMgG])?'
+        m = re.match(regex, s)
+        if m:
+            suffix = m.group(2)
+            if suffix is None:
+                return HeapSize(int(m.group(1)), "")
+            else:
+                return HeapSize(int(m.group(1)), suffix.upper())
+        else:
+            raise Exception("Invalid Heap Size {}!".format(s))
```

Comment:
A user may question what is a suitable heap size if they get an error. Exposing the actual RegEx would be one approach:

```suggestion
            raise Exception("Invalid Heap Size format '{}'! Format should be: '{}'.".format(s, regex))
```